### PR TITLE
Add a cache layer to Docker

### PR DESCRIPTION
**Pull request by mdashti**
This PR adds a cache later to Docker using `cargo-chef`. This change has a noticeable improvement in the turnaround time if someone is using Docker in their development cycle.

### CODE REVIEW

**Comment by levkk** [comment]
Not too familiar with `cargo chef`. Is it ok to include this in the production image? If not, we can always create a `Dockerfile.dev`.

File: Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder 
+COPY --from=planner /app/recipe.json recipe.json
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --recipe-path recipe.json
+# Build application
+COPY . .
+RUN cargo build --release
```
Comment (levkk) [inline Dockerfile.dev:18]: You may want a debug build here.

**Comment by mdashti** [comment]
@levkk `cargo chef` is definitely the accepted docker cache layer, but I'm also not sure about its production grade. From their repo:

```
cargo-chef was initially developed for the deployment chapter of [Zero to Production In Rust](https://zero2prod.com/), a hands-on introduction to backend development using the Rust programming language.
```

Which might mean initially was not meant to be for production, but it is now.

Anyways, your suggestion makes sense. Thanks.

File: Dockerfile.dev (hunk 1 above)
Comment (mdashti) [reply Dockerfile.dev]: Thanks. Done.

**Review by levkk — APPROVED** [review]
LGTM!

**Pull request merged**